plugin/evm/orderbook/contract_events_processor.go

```diff
@@ -704,17 +705,30 @@ func (cep *ContractEventsProcessor) updateMetrics(logs []*types.Log) {
 
 		metricName := fmt.Sprintf("%s/%s", "events", event_.Name)
 
-		if !event.Removed {
-			metrics.GetOrRegisterCounter(metricName, nil).Inc(1)
-		} else {
-			metrics.GetOrRegisterCounter(metricName, nil).Dec(1)
-		}
+		metrics.GetOrRegisterCounter(metricName, nil).Inc(1)
 
 		switch event_.Name {
 		case "OrderAccepted":
-			orderAcceptedCount++
+			orderAcceptedCount += 1
 		case "OrderCancelAccepted":
-			orderCancelledCount++
+			orderCancelledCount += 1
+		case "OrderMatchingError":
+			// separate metrics for combination of order type and error string - for more granular analysis
+			args := map[string]interface{}{}
+			err := cep.orderBookABI.UnpackIntoMap(args, "OrderMatchingError", event.Data)
+			if err != nil {
+				log.Error("error in orderBookAbi.UnpackIntoMap", "method", "OrderMatchingError", "err", err)
+				return
+			}
+			orderId := event.Topics[1]
+			errorString := args["err"].(string)
+
+			order := cep.database.GetOrderById(orderId)
+			if order != nil {
```

Comment:
if it is nill, it's deleted from the db for some reason. Ideally shouldn't happen but should we atleast log it?

Reply:
one possible reason when it could happen is how currently there are OrderMatchingErrors, and it's possible that by the time those blocks are accepted the orders are already deleted cuz they were matched